File: CareSync.BusinessAPI/Entities/MedicalRecord.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace CareSync.BusinessAPI.Entities;

public class MedicalRecord : BaseEntity
{
    public Guid AppointmentId { get; set; }

    public DateTime VisitDate { get; set; }

    public string Diagnosis { get; set; } = string.Empty;

    public string Symptoms { get; set; } = string.Empty;

    public string Treatment { get; set; } = string.Empty;

    public string DoctorNotes { get; set; } = string.Empty;

    [ForeignKey(nameof(AppointmentId))]
    public Appointment Appointment { get; set; } = null!;
}

File: frontend/src/types/MedicalRecord.ts
export interface MedicalRecord {
  id: string;

  appointmentId: string;

  patientName: string;

  doctorName: string;

  department: string;

  visitDate: string;

  diagnosis: string;

  symptoms: string;

  treatment: string;

  doctorNotes: string;
}

export interface CreateMedicalRecordRequest {
  appointmentId: string;

  visitDate: string;

  diagnosis: string;

  symptoms: string;

  treatment: string;

  doctorNotes: string;
}

export interface UpdateMedicalRecordRequest {
  visitDate: string;

  diagnosis: string;

  symptoms: string;

  treatment: string;

  doctorNotes: string;
}